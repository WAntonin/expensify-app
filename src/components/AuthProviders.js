import React from 'react'
import { connect } from 'react-redux'
import { startGoogleLogin, startFacebookLogin } from '../actions/auth'

export class AuthProviders extends React.Component {
    render() {
        return (
            <div >
                <button className="button button--login" onClick={this.props.startGoogleLogin}>
                    <img className="button--login__item" src="/images/google_icon.png" />
                    <span>Sign in with Google</span>
                </button>
                <button className="button button--login" onClick={this.props.startFacebookLogin}>
                    <img className="button--login__item" src="/images/facebook_icon.png" />
                    <span>Sign in with Facebook</span>
                </button>
            </div>
        )
    }
}

const mapDispatchToProps = (dispatch) => ({
    startGoogleLogin: () => dispatch(startGoogleLogin()),
    startFacebookLogin: () => dispatch(startFacebookLogin())
})

export default connect(undefined, mapDispatchToProps)(AuthProviders)