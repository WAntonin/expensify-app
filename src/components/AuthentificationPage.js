import React from 'react'
import { connect } from 'react-redux'

export class AuthentificationPage extends React.Component {
    render() {
        return (
            <div className="box-layout">
                <div className="box-layout__box">
                    <h1 className="box-layout__title">Expensify App</h1>
                    <h2 className="box-layout__subtitle">An application to log your expenses</h2>
                </div>
            </div>
        )
    }
}