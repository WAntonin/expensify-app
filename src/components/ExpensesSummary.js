import React from 'react'
import { connect } from 'react-redux'
import selectExpenses from '../selectors/expenses'
import selectTotalExpenses from '../selectors/expenses-total'
import numeral from 'numeral'


export const ExpensesSummary = ({expenseCount, expenseTotal}) => {
    const plural = expenseCount > 1 ? 's' : ''
    const formattedExpenseTotal = numeral(expenseTotal / 100).format('0,0.00$')
    return(
    <div>
        <h1>Viewing {expenseCount} expense{plural} for a total of {formattedExpenseTotal}</h1>
    </div>
    )
}

const mapStateToProps = (state) => {
    const filteredExpenses = selectExpenses(state.expenses, state.filters)
    return {
        expenseCount: filteredExpenses.length,
        expenseTotal: selectTotalExpenses(filteredExpenses)
    }
}

export default connect(mapStateToProps)(ExpensesSummary)