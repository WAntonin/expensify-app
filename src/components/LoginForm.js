import React from 'react'
import isEmail from 'validator/lib/isEmail'
import { startEmailPasswordLogin, startGoogleLogin, startFacebookLogin } from '../actions/auth'
import { connect } from 'react-redux';

export class LoginForm extends React.Component {
    constructor(props) {
        super(props)
        this.state = {
            email: '',
            password: '',
            error: ''
        }
    }
    onChange = (e) => {
        this.setState({ [e.target.name]: e.target.value })
    }
    onSubmit = (e) => {
        e.preventDefault()
        if (!isEmail(this.state.email) || this.state.password.length < 6) {
            this.setState(() => ({ error: 'Invalid password or email' }))
        } else {
            this.setState(() => ({ error: '' }))
            this.props.startEmailPasswordLogin(this.state.email, this.state.password)
        }
    }
    render() {
        const isInvalid = this.state.password === '' || this.state.email === ''
        return (
            <div className="box-layout__box">
                <button className="button button--login" onClick={this.props.startGoogleLogin}>
                    <img className="button--login__item" src="/images/google_icon.png" />
                    <span>Sign in with Google</span>
                </button>
                <button className="button button--login" onClick={this.props.startFacebookLogin}>
                    <img className="button--login__item" src="/images/facebook_icon.png" />
                    <span>Sign in with Facebook</span>
                </button>
                <h3>OR</h3>
                <form onSubmit={this.onSubmit}>
                    <input
                        type="email"
                        value={this.state.email}
                        name="email"
                        onChange={this.onChange}
                        placeholder="email"
                        autoComplete="on"
                    />
                    <input
                        type="password"
                        value={this.state.password}
                        name="password"
                        onChange={this.onChange}
                        placeholder="password (6 char. min.)"

                        autoComplete="on"
                    />
                    <button disabled={isInvalid} type="submit">Sign In</button>
                </form>
            </div>
        )
    }
}

const mapDispatchToProps = (dispatch) => ({
    startGoogleLogin: () => dispatch(startGoogleLogin()),
    startFacebookLogin: () => dispatch(startFacebookLogin()),
    startEmailPasswordLogin: (email, password) => dispatch(startEmailPasswordLogin(email, password))
})

export default connect(undefined, mapDispatchToProps)(LoginForm)