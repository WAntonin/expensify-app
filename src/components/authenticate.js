import React from 'react'

const authenticate = (WrappedComponent) => {
    class Authenticate extends React.Component {
        constructor(props) {
            super(props)
            this.state = {
                email: '',
                password: ''
            }
            this.onInputChange = this.onInputChange.bind(this);
            this.submit = this.submit.bind(this)
        }
        onInputChange = (e) => {
            e.preventDefault()
            console.log('INPUT_CHANGE')
            const inputName = e.target.name 
            const inputValue = e.target.value 
            this.setState({ [inputName]: inputValue})
        }
        submit = () => {
            console.log('SUBMIT')
            this.props.onSubmit({
                email: this.state.email,
                password: this.state.password
            })
            console.log('SUBMIT', this.state.email, this.state.password)
            debugger
        }
        render() {
            const { onChange, onSubmit, ...otherProps } = this.props
            return (
                <WrappedComponent
                onChange={ this.onInputChange}
                onSubmit={this.submit}
                {...otherProps}
                />)
        }
    }
    return Authenticate
}

export default authenticate