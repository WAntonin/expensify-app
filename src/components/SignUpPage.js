import React from 'react'
import { connect } from 'react-redux'
import SignUpForm from './SignUpForm'
import AuthProviders from './AuthProviders'
import { startCreateUserAccount } from '../actions/auth'

export class SignUpPage extends React.Component {
    onSubmit = ({ email, password }) => {
        this.props.startCreateUserAccount(email, password)
    }
    render() {
        return (
            <div className="box-layout">
                <div className="box-layout__box">
                    <AuthProviders />
                    <h3>OR</h3>
                    <SignUpForm onSubmit={this.onSubmit} />
                </div>
            </div>
        )
    }
}

const mapStateToProps = (state) => ({
    user: state.user,
    email: state.email,
    password: state.password
})

const mapDispatchToProps = (dispatch) => ({
    startCreateUserAccount: (email, password) => dispatch(startCreateUserAccount(email, password))
})

export default connect(mapStateToProps, mapDispatchToProps)(SignUpPage)