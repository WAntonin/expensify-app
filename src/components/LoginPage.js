import React from 'react'
import LoginForm from './LoginForm'
import { Link } from 'react-router-dom'
import { startEmailPasswordLogin } from '../actions/auth'
import { connect } from 'react-redux'
import AuthProviders from './AuthProviders';

export class LoginPage extends React.Component {
    onSubmit = ({ email, password }) => {
        this.props.startEmailPasswordLogin(email, password).catch((error) => {
            this.setState({ error })
        })
    }
    render() {
        return (
            <div className="box-layout">
                <div className="box-layout__box">
                    <h1 className="box-layout__title">Expensify App</h1>
                    <p>An application to log your expenses</p>
                </div>
                <div className="box-layout__box">
                    <AuthProviders />
                    <h3>OR</h3>
                    <LoginForm onSubmit={this.onSubmit} />
                    {error.code === 'auth/wrong-password' && <p>Forgot your password?</p>}
                </div>
                <Link className="box-layout__box box-layout__box--link" to="/signup">
                    <p>If you don't have an account you can sign up here.</p>
                </Link>
            </div>
        )
    }
}

const mapStateToProps = (state) => ({
    email: state.email,
    password: state.password,
    error: state.error
})

const mapDispatchToProps = (dispatch) => ({
    startEmailPasswordLogin: (email, password) => dispatch(startEmailPasswordLogin(email, password))
})

export default connect(mapStateToProps, mapDispatchToProps)(LoginPage)