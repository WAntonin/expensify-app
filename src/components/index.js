import React from 'react'
import { connect } from 'react-redux'
import AuthProviders from './AuthProviders'
import SignIn from './SignIn'
import SignUp from './SignUp'
import { startEmailPasswordLogin, startCreateUserAccount } from '../actions/auth'
import authenticate from './authenticate'

const SignInForm = authenticate(SignIn)
const SignUpForm = authenticate(SignUp)


export class Authenticate extends React.Component {
    constructor(props) {
        super(props)
        this.state = {
            isSignIn: true
        }
        this.toggleForm = this.toggleForm.bind(this)
    }
    toggleForm = () => {
        this.setState(() => ({
            isSignIn: !this.state.isSignIn
        }))
    }
    render() {
        const { startCreateUserAccount, startEmailPasswordLogin } = this.props
        return (
            // <div className="box-layout">
            //     <div className="box-layout__box">
            //         <h1 className="box-layout__title">Expensify App</h1>
            //         <h2 className="box-layout__subtitle">An application to log your expenses</h2>
            //     </div>
            // </div>
            <div className="box-layout">
                <div className="box-layout__box">
                    <AuthProviders />
                </div>
                {
                    this.state.isSignIn ? (
                        <SignInForm
                            onSubmit={startEmailPasswordLogin}
                            showSignUp={this.toggleForm}
                        />
                    ) : 
                    (
                        <SignUpForm
                            onSubmit={startCreateUserAccount}
                            showSignIn={this.toggleForm}
                        />
                    )
                }
            </div>
        )
    }
}

const mapDispatchToProps = (dispatch) => ({
    startCreateUserAccount: (email, password) => dispatch(startCreateUserAccount(email, password)),
    startEmailPasswordLogin: (email, password) => dispatch(startEmailPasswordLogin(email, password))
})

export default connect(undefined, mapDispatchToProps)(Authenticate)