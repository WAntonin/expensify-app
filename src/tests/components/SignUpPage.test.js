import React from 'react'
import { shallow } from 'enzyme'
import { SignUpPage } from '../../components/SignUpPage'
import userId from '../fixtures/userId.js'

test('should render SignUpPage correctly', () => {
    const wrapper = shallow(<SignUpPage />)
    expect(wrapper).toMatchSnapshot()
})

test('should handle startCreateUserAccount on SignUpForm submit', () => {
    const startCreateUserAccount = jest.fn()
    const wrapper = shallow(<SignUpPage
        userId={userId}
        startCreateUserAccount={startCreateUserAccount}
        />)
    wrapper.find('SignUpForm').prop('onSubmit')(userId)
    expect(startCreateUserAccount).toHaveBeenLastCalledWith(userId.email, userId.password)
})