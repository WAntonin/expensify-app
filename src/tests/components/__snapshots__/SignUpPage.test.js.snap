// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render SignUpPage correctly 1`] = `
<div
  className="box-layout"
>
  <div
    className="box-layout__box"
  >
    <Connect(AuthProviders) />
    <h3>
      OR
    </h3>
    <SignUpForm
      onSubmit={[Function]}
    />
  </div>
</div>
`;
